cmd/bad-key-revoker/main.go

```diff
@@ -110,9 +110,10 @@ func (bkr *badKeyRevoker) findUnrevoked(unchecked uncheckedBlockedKey) ([]unrevo
 		}
 		_, err := bkr.dbMap.Select(
 			&batch,
-			"SELECT id, certserial FROM keyHashToSerial WHERE keyHash = ? AND id > ? ORDER BY id LIMIT ?",
+			"SELECT id, certSerial FROM keyHashToSerial WHERE keyHash = ? AND id > ? AND certNotAfter > ? ORDER BY id LIMIT ?",
 			unchecked.KeyHash,
 			initialID,
+			time.Now(),
```

Comment:
I see the rest of this test is using `time`, but generally Boulder prefers `"github.com/jmhodges/clock"`, which in this case you'd add to the `badKeyRevoker` struct. The default implementation is time as you expect, but for the tests, you can override the clock so that you don't have to do clock math. See https://github.com/letsencrypt/boulder/blob/0be4679a9b2e06e6bc553ce5853686c9f846f0d1/sa/sa.go#L38 and https://github.com/letsencrypt/boulder/blob/0be4679a9b2e06e6bc553ce5853686c9f846f0d1/sa/sa_test.go#L52-L53

Reply:
Seconding this recommendation. You'll add
```go
type badKeyRevoker struct {
    ...
    clk   clock.Clock
}
```
at the top of this file. Then in `main()` you'll set
```go
bkr := &badKeyRevoker{
    ...
    clk: cmd.Clock()
}
```
and in `main_test.go` you'll create a
```go
fc := clock.NewFake()
```
and pass it around (e.g. into the `insertCert` function), using it everywhere the tests currently use `time.Foo`.